### PR TITLE
Release development to main

### PR DESCRIPTION
**Pull request by dimileeh**
## Summary

Promote the current development branch to main.

Includes:
- F21V Core PR monitor runtime execution seam for GKE (#751)

This release introduces a cloud-neutral hosted execution seam so PR monitor repair and other agent runs can execute through an injected runtime executor while local Core remains unchanged when no executor is configured. It also expands hosted credential-name passthrough, Compose environment interpolation and redaction handling, hosted failure classification, and recovery behavior that preserves hosted errors while still starting Compose services for validation.

## Validation

AWF/GitHub CI own the broad merge-gating validation for this release PR, including comments, checks, freshness, validation provenance, and merge policy. No additional broad validation was run manually in this metadata-only update.

## Release Readiness

Affected AWF Core surfaces include agent adapters, executor wiring, PR monitor recovery, profile/Compose environment handling, service worker handoff paths, and related unit coverage. Hosted execution remains opt-in through the injected runtime executor; Compose remains the default local Core path.

## Security / Secrets

No secrets were added to code, logs, fixtures, screenshots, or this PR description. The release specifically keeps hosted credential transport secret-free by passing credential names for out-of-band resolution and redacting secret-like profile environment values.

---

> [!NOTE]
> **High Risk**
> Large change surface across agent execution, credential/env transport, and monitor recovery semantics; misclassification or secret leakage on the hosted path would affect production repair runs.
> 
> **Overview**
> Introduces a **cloud-neutral `AgentRuntimeExecutor` seam** so agent runs can delegate to an injected backend (e.g. GKE Jobs) instead of `docker compose exec` when configured; local Core is unchanged when no executor is set. `AgentAdapter` builds secret-free `AgentRuntimeExecRequest` payloads (credential **names**, aliases, file-auth mount targets, literal `profile_env`) with optional live log streaming, wall/idle watchdogs, and the same failure classification as Compose.
> 
> **Compose/profile handling** is expanded for hosted parity: filtering of passthrough names against profile-owned slots, `literal_profile_env_from_compose`, GitHub token source/alias surfacing, ADC mount targets, and stricter secret/credential redaction—mostly refactored into `compose_auth_env` / `compose_env` / related helpers.
> 
> Per-adapter **`hosted_env_passthrough_names`** contracts align hosted auth with `AGENT_AUTH_ENV_VARS`. **Grok** moves from plain `grok -p` to an **ACP `grok agent stdio` bridge** so prompts stay on stdin.
> 
> **Executor/monitor paths** pass `runtime_executor` into adapters and **bypass Compose agent-service recovery** when `adapter.is_hosted`, while monitor resume still validates compose stack readiness (including expected services after failed restarts). **Agent-runtime image** pins newer Codex/Claude/Gemini/OpenCode/Grok versions and adds build-time CLI smoke checks.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c926b2df1ec33932db486680b823dd99deacdfff. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added hosted agent execution support via an injected runtime executor, including structured request/response handling and optional live stdout/stderr streaming.
  * Introduced adapter-specific “hosted passthrough” allowlists so hosted runs receive only safe environment names (not secret values).
  * Enhanced hosted GitHub token handling to correctly select sources and alias names.
* **Bug Fixes**
  * Prevented hosted timeouts/failures from being misclassified as local compose agent recovery issues.
  * Hardened Compose-derived hosted profile environment handling with stricter secret/credential redaction (including Postgres passwords).
* **Tests**
  * Expanded unit coverage for hosted execution, credential passthrough contracts, secret-leak prevention, and hosted recovery/timeout edge cases.

### CODE REVIEW

**Comment by dimileeh** [comment]
⚠️ PR #754 needs human attention at commit `6bd01e4621`.

AWF did not auto-merge because GitHub reports merge state BLOCKED; required protection or review hooks need a human.

After the blocker is cleared or a new commit lands, AWF will re-verify the PR before taking any merge action.

File: src/awf/control/executor/monitor_handoff.py [hunk 1]
```diff
@@ -494,7 +525,124 @@ async def _compose_runtime_usable_after_restart_failure(compose_project: str) ->
...
+    if not isinstance(payload, dict):
+        return _ComposeRuntimeRequirements(
+            service_names=set(), successful_completion_services=set()
+        )
+    services = payload.get("services")
+    if not isinstance(services, dict):
+        return _ComposeRuntimeRequirements(
+            service_names=set(), successful_completion_services=set()
+        )
+    service_names = {name for name in services if isinstance(name, str) and name}
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/control/executor/monitor_handoff.py]: Fixed in 695f2eba1. The resume fallback now derives required services from the same COMPOSE_PROFILES selection used by docker compose up: services without profiles remain required, inactive profiled services are excluded, and explicitly active or wildcard-selected services remain required. Successful one-shot dependency handling is also computed only from active service definitions. Added regression coverage for inactive, unrelated, active, wildcard, malformed, and ordinary service cases; the focused suite, Ruff, mypy, maintainability checks, and isolated wheel smoke pass.

File: src/awf/profiles/compose_git_config.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    from awf.profiles import compose as compose_module
+
+    carried_entries: list[tuple[str, str | None, str | None]] = []
+    for index in range(count):
+        config_key_raw = compose_env.get(f"{_GIT_CONFIG_KEY_PREFIX}{index}")
+        config_value_raw = compose_env.get(f"{_GIT_CONFIG_VALUE_PREFIX}{index}")
+        if (
+            config_key_raw is None
+            or config_value_raw is None
+            or config_key_raw == _COMPOSE_PASSTHROUGH
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/profiles/compose_git_config.py]: Fixed in c926b2df1. Pass-through GIT_CONFIG_KEY_n slots now resolve the matching worker key before the existing URL-userinfo, safe SSH rewrite, and Bitbucket rewrite filters run. Regression coverage keeps a safe credential.helper entry, rejects a worker key containing URL credentials, skips a missing source, and verifies reindexing.

File: src/awf/profiles/compose_auth_env.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+_NETRC_AUTH_CREDENTIAL_LIKE_VALUE_PATTERN = re.compile(
+    r"(?:^|[\r\n])\s*(?:machine[^\S\r\n]+\S+|default)(?:[^\S\r\n]+\S+)*"
+    r"[^\S\r\n]+password[^\S\r\n]+\S+",
+    re.IGNORECASE,
+)
+_URL_SECRET_CREDENTIAL_FIELD_NAMES = frozenset(
+    {"JWT", "PASSWORD", "PASSWD", "SECRET", "SIGNATURE", "TOKEN"}
+)
+_URL_SECRET_CREDENTIAL_FIELD_EXACT_NAMES = frozenset(
+    {
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/profiles/compose_auth_env.py:256]: Fixed in c926b2df1. SAMLRESPONSE and SAMLASSERTION are now exact credential-bearing URL field names. Regression coverage verifies both query spellings are detected and that neutral APP_URL/CALLBACK_URL values containing assertions are excluded from hosted profile_env while an ordinary PUBLIC_URL remains.

File: src/awf/profiles/compose_git_config.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+def _hosted_git_config_env(
+    compose_env: Mapping[str, str],
+    *,
+    worker_env: Mapping[str, str],
+    skip_bitbucket_agent_rewrites: bool,
+) -> tuple[tuple[tuple[str, str], ...], tuple[tuple[str, str], ...]]:
+    count_raw = compose_env.get(_GIT_CONFIG_COUNT_KEY)
+    if count_raw is None or count_raw == _COMPOSE_PASSTHROUGH:
+        return (), ()
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/profiles/compose_git_config.py]: Fixed in c926b2df1. Pass-through GIT_CONFIG_COUNT now resolves the worker-provided value, validates it as an integer, and rebuilds the hosted block from only retained safe entries. Regression coverage verifies the pass-through count preserves credential.helper configuration and that a missing worker count still fails closed.

File: src/awf/profiles/compose_git_config.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+    count_value, count_resolution = _compose_resolve_value(count_raw, worker_env=worker_env)
+    if count_resolution is not _ComposeEnvResolution.LITERAL:
+        return (), ()
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/profiles/compose_git_config.py]: Fixed in c926b2df1. Worker-selected GIT_CONFIG_COUNT expressions now resolve their selected worker source, validate the value as an integer, and preserve the safe hosted Git config block. Regression coverage exercises ${GIT_CONFIG_COUNT}=1 plus missing-source and non-integer fail-closed cases.
